pkg/endpoint/endpoint.go

```diff
@@ -2268,6 +2268,7 @@ func (e *Endpoint) ModifyIdentityLabels(owner Owner, addLabels, delLabels pkgLab
 
 	switch e.GetStateLocked() {
 	case StateDisconnected, StateDisconnecting:
+		e.Unlock()
```

Comment:
Can you push this into a separate PR? This warrants backporting.

Reply:
Done (#6145)